snapcraft/config.py

```diff
@@ -25,16 +25,60 @@
 
 from xdg import BaseDirectory
 
-from snapcraft.storeapi import (
-    constants,
-    errors,
-)
+from snapcraft.internal.errors import SnapcraftInvalidCLIConfigError
+from snapcraft.storeapi import constants, errors
 
 LOCAL_CONFIG_FILENAME = '.snapcraft/snapcraft.cfg'
 
 logger = logging.getLogger(__name__)
 
 
+class CLIConfig:
```

Comment:
This class is really just a dict that saves keys and values to to a file. I'm not a huge fan, for the same reason I don't like accessing the well-defined snapcraft.yaml as a dict: one typo, and you're getting the wrong thing. Want to change the key name? You need to change it in every place you use it. We have the same problem with needing to remember that we need to use `'Sentry'` and `'always_send'`. Why not encapsulate that knowledge here and provide a specific API for setting/getting that property?

Reply:
sure

Reply:
I am plus one on this, just didn't want to move away from the status quo too much :-)